flang/lib/Lower/CharacterExpr.cpp

```diff
@@ -615,3 +615,17 @@ bool Fortran::lower::CharacterExprHelper::isArray(mlir::Type type) {
   }
   return false;
 }
+
+bool Fortran::lower::CharacterExprHelper::hasConstantLengthInType(
+    const fir::ExtendedValue &exv) {
+  auto type = fir::getBase(exv).getType();
+  if (auto refType = type.dyn_cast<fir::ReferenceType>())
+    type = refType.getEleTy();
+  if (auto seqTy = type.dyn_cast<fir::SequenceType>()) {
+    auto eleTy = seqTy.getEleTy();
+    auto lenVal = seqTy.getShape()[0];
+    return lenVal != fir::SequenceType::getUnknownExtent() &&
+           eleTy.isa<fir::CharacterType>();
```

Comment:
Seems like if we get here and this isn't even a CHARACTER, something is broken really bad. Shouldn't this test be an `assert` or even `emitError`?

Reply:
Right, that would be a broken case and their are no good reason to accept it. I have added an `assert`.